special-pages/pages/new-tab/app/favorites/components/Favorites.js

```diff
@@ -265,10 +271,20 @@ function useVisibleRows(rows, rowHeight, safeAreaRef) {
         }
         const startIndex = Math.floor(start / rowHeight);
         const endIndex = Math.min(Math.ceil(end / rowHeight), rows.length);
-        setVisibleRange({ start: startIndex, end: endIndex });
+
+        // don't set state if the offset didn't change
+        setVisibleRange((prev) => {
+            if (startIndex !== prev.start || endIndex !== prev.end) {
+                return { start: startIndex, end: endIndex };
+            }
+            return prev;
+        });
```

Comment:
preact/react will avoid the cascade of updating components if you return a stable reference to the old values. Since this was firing on scroll, it's a worthwhile optimization to prevent work